vm.go

```diff
@@ -731,3 +749,71 @@ func getSchemaVersion(opts *VirtualMachineOptions) hcsschema.Version {
 		Minor: 1,
 	}
 }
+
+func (vm *VirtualMachineSpec) GetDiskInformation(diskPath string) (controlNumber int, err error) {
+	for key, value := range vm.spec.VirtualMachine.Devices.Scsi["primary"].Attachments {
+		if value.Path == diskPath {
+			controlNumber, err = strconv.Atoi(key)
+			return
+		}
+	}
+
+	return 0, errors.New("diskpath not found in attached devices")
+}
+
+func (vm *VirtualMachineSpec) AttachVhd(vhdPath string) (lun int, err error) {
+	ctx, cancel := context.WithTimeout(context.Background(), 60*time.Minute)
+	defer cancel()
+	system, err := hcs.OpenComputeSystem(ctx, vm.ID)
+	if err != nil {
+		return 0, err
+	}
+	defer system.Close()
+
+	if err := wclayer.GrantVmAccess(vm.ID, vhdPath); err != nil {
+		return 0, err
+	}
+
+	for lun := 0; lun < 256; lun++ {
```

Comment:
>lun [](http://example.com/codeflow?start=15&length=3)

why attach the same vhd from lun 0?  We should find the free lun and attach the vhd using the free lun.

Reply:
as far as i can tell there is no way to find the next free lun though any hcs/hcsshim calls. unless we want to make a sperate service to track the lun for each vm we create, we will have to get it through trial and error. The spec will not help us in this case since it is not stored anywhere and will not be populated if we call the GetVirtualMachineSpec function to retrieve the VirtualMachineSpec struct. I can start the lun count at 2 since 0 and 1 are required for creation.